packages/nimble-components/src/table/index.ts

```diff
@@ -760,13 +803,29 @@ export class Table<
             );
         }
         if (this.tableUpdateTracker.requiresTanStackDataReset) {
```

Comment:
I updated this to also reset for `updateRowIds` but not `updateRowParentIds`, because I don't think that technically conflict with the current expanded state. I could definitely be convinced otherwise.

Reply:
If I'm thinking about this correctly, then I think it would be good to reset the expanded state when parent row IDs changes since: (1) I think that will be necessary when we have lazy-loaded hierarchy, (2) I'm not sure if makes sense from a user's perspective to keep some rows expanded when they are hierarchically ordered in a different way, and (3) the set of rows that `isRowExpandable` may change, which might cause issues with the expansion manager (we'd have to think through if there really are problems here).